test/model.test.js

```diff
@@ -0,0 +1,893 @@
+
+    doc.str = doc.str + util.s8();
+    expect(doc.str).to.not.eql(otherDoc.str);
+
+    var anotherDoc = new test.models.Model(values);
+    expect(anotherDoc).to.not.eql(otherDoc);
+    expect(anotherDoc).to.not.eql(doc);
+  });
+
+  it('should create two distinct instances with the same options', function() {
+    var options = { str: util.s8() };
+    var doc1 = new test.models.Model(options);
+    var doc2 = new test.models.Model(options);
+
+    expect(doc1).to.not.eql(doc2);
+  });
+
+  it('should ensure two instances are different', function() {
+    var str1 = util.s8(), str2 = util.s8();
+    var doc1 = new test.models.Model({ str: str1 });
+    expect(doc1.str).to.equal(str1);
+
+    var doc2 = new test.models.Model({ str: str2 });
+    expect(doc2.str).to.equal(str2);
+    expect(doc1.str).to.equal(str1);
+    expect(doc1).to.not.eql(doc2);
+  });
+
+  it('two instances should have different prototypes', function() {
+    var doc1 = new test.models.Model({ str: util.s8() });
+    var doc2 = new test.models.Model({ str: util.s8() });
+
+    expect(Object.getPrototypeOf(doc1)).to.not.eql(Object.getPrototypeOf(doc2));
+  });
+
+  it('two instances should have the same model', function() {
+    var doc1 = new test.models.Model({ str: util.s8() });
+    var doc2 = new test.models.Model({ str: util.s8() });
+
+    expect(doc1.getModel()).to.eql(doc2.getModel());
+  });
+
+  it('docs from different models should not interfere', function() {
+    var str = util.s8();
+    var doc = new test.models.Model({str: str});
+
+    var otherName = util.s8();
+    test.models.OtherModel = thinky.createModel(otherName, {
+      str: String
+    });
+
+    return test.models.OtherModel.ready()
+      .then(function() {
+        var otherStr = util.s8();
+        var otherDoc = new test.models.OtherModel({ str: otherStr });
+
+        expect(doc.str).to.eql(str);
+        expect(otherDoc.str).to.eql(otherStr);
+
+        expect(otherDoc.getModel()).to.not.eql(doc.getModel());
+
+        // TODO: better way to access schema data, maybe Model.getSchema()
+        expect(doc.getModel().getTableName()).to.eql(test.models.Model._schema._model._name);
+        expect(otherDoc.getModel().getTableName(), otherName);
+      });
+  });
+
+});
+
+describe("Batch insert", function() {
+  afterEach(function() { return test.dropTables(); });
+
+  it('should insert a single document', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number
+    });
+
+    return Model.ready()
+      .then(function() {
+        return Model.save({ id: 'foo' });
+      })
+      .then(function(result) {
+        // NOTE: deepCopy to remove prototype. Otherwise these are
+        //       not, in fact, equivalent objects, even though
+        //       assert will claim they are
+        expect(util.deepCopy(result)).to.eql({ id: 'foo' });
+      });
+  });
+
+  it('should insert a batch of documents', function() {
```

Comment:
It's missing `done` here. The test is synchronous. If the test fails, it may print something like "Uncaugh Exception...`, but the test will still pass